src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -107,6 +108,26 @@ def _build_general_name(backend, gn):
         return x509.DirectoryName(
             _build_x509_name(backend, gn.d.directoryName)
         )
+    elif gn.type == backend._lib.GEN_EMAIL:
```

Comment:
This seems reasonable to me.

On Sat, May 2, 2015 at 4:15 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/x509.py
> <https://github.com/pyca/cryptography/pull/1881#discussion_r29550321>:
>
> > @@ -107,6 +108,26 @@ def _build_general_name(backend, gn):
> >          return x509.DirectoryName(
> >              _build_x509_name(backend, gn.d.directoryName)
> >          )
> > +    elif gn.type == backend._lib.GEN_EMAIL:
>
> This is the last branch to add and I wouldn't anticipate further changes
> for now. We can refactor if we have reason to revisit this though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1881/files#r29550321>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084